src/Application.vue

```diff
@@ -312,22 +312,27 @@ export default {
         // (Widget mode is on when provider is initialized from a widget instance)
         if (state.widgetMode) {
           let widget = _.find(this.savedFields.widgetInstances, { id: state.widgetId });
+          const component = state.widgetData.widgetLayout
+            ? widgetsMap[`${state.widgetData.widgetPackage}:${state.widgetData.widgetLayout}`]
+            : state.widgetData.component || widgetsMap[state.widgetData.widgetPackage];
```

Comment:
@tonytlwu combining ternary with `||` reduces readability. In such cases, when I want to provide fallback value to ternary, I would rather stick to logical operators:

```
const component = state.widgetData.widgetLayout && widgetsMap[`${state.widgetData.widgetPackage}:${state.widgetData.widgetLayout}`]
            || state.widgetData.component
            || widgetsMap[state.widgetData.widgetPackage];
```